src/main/java/org/veupathdb/service/eda/us/service/UserService.java

```diff
@@ -90,7 +84,22 @@ public PatchUsersAnalysesByUserIdAndProjectIdAndAnalysisIdResponse patchUsersAna
     User user = Utils.getAuthorizedUser(_request, userId);
     AnalysisDetailWithUser analysis = dataFactory.getAnalysisById(analysisId);
     Utils.verifyOwnership(user.getUserID(), analysis);
+
+    var oldDescriptor = analysis.getDescriptor();
+
     editAnalysis(user, analysis, entity);
+
+    // If the descriptor instance has changed AND we have derived variables then
+    // validate the derived variables to ensure all the IDs point to actual
```

Comment:
Maybe move this comment (or copy the portion saying what validation steps we take) down to the method where it's happening.  Kept having to peek up here to see what the validation method was trying to do.